sql/core/src/main/scala/org/apache/spark/sql/jdbc/MySQLDialect.scala

```diff
@@ -94,4 +94,23 @@ private case object MySQLDialect extends JdbcDialect {
   override def getTableCommentQuery(table: String, comment: String): String = {
     s"ALTER TABLE $table COMMENT = '$comment'"
   }
+
+  override def getJDBCType(dt: DataType): Option[JdbcType] = dt match {
+    case IntegerType => Option(JdbcType("INTEGER", java.sql.Types.INTEGER))
```

Comment:
I think you can remove other pattern matches and delegate it to `getCommonJDBCType`:

https://github.com/apache/spark/blob/0494dc90af48ce7da0625485a4dc6917a244d580/sql/core/src/main/scala/org/apache/spark/sql/jdbc/PostgresDialect.scala#L94-L95

So to clarify we need to extra pattern matches, is it correct?

```scala
    case DoubleType => Option(JdbcType("REAL", java.sql.Types.DOUBLE))
    case FloatType => Option(JdbcType("FLOAT", java.sql.Types.FLOAT))
```

Reply:
Thanks for the comments. I think we need just one pattern match: 
`case FloatType => Option(JdbcType("FLOAT", java.sql.Types.FLOAT))`

That's because `DoubleType` is correctly mapped to `DOUBLE PRECISION` in `getCommonJDBCType`. WDYT?

Reply:
I thought you commented that `REAL` is treated as double in MySQL. I don't see REAL is being mapped at `getCommonJDBCType`. Should we add this too?

Reply:
In https://github.com/apache/spark/blob/0494dc90af48ce7da0625485a4dc6917a244d580/sql/[%E2%80%A6]org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala there is a mapping between `DoubleType` and `DOUBLE PRECISION`. Given this, I think we don't need to add a mapping between `DoubleType` and `REAL` in the MySQLDialect overridden method.